File: mot-selenium/src/main/java/com/dvsa/mot/selenium/datasource/DvlaVehicle.java
package com.dvsa.mot.selenium.datasource;

import java.util.Date;

public class DvlaVehicle {

    public static final DvlaVehicle DVLA_VEHICLE_WITH_VALID_MAKE_AND_MODEL =
            new DvlaVehicle("29F29", "10D0E");

    private String registration;
    private String registrationValidationCharacter;
    private String vin;
    private String makeCode;
    private String modelCode;
    private String makeInFull;
    private String colour1Code;
    private String colour2Code;
    private String propulsionCode;
    private Integer designedGrossWeight;
    private Integer unladenWeight;
    private String engineNumber;
    private Integer engineCapacity;
    private Integer seatingCapacity;
    private Date manufactureDate;
    private Date firstRegistrationDate;
    private Boolean isSeriouslyDamaged;
    private String recentV5DocumentNumber;
    private Boolean isVehicleNewAtFirstRegistration;
    private String bodyTypeCode;
    private String wheelPlanCode;
    private String svaEmissionStandard;
    private String ctRelatedMark;
    private Integer vehicleId;
    private Integer dvlaVehicleId;
    private String euClassification;
    private Integer massInServiceWeight;


    public DvlaVehicle(String makeCode, String modelCode) {
        this.makeCode = makeCode;
        this.modelCode = modelCode;
    }


    public String getRegistration() {
        return registration;
    }

    public void setRegistration(String registration) {
        this.registration = registration;
    }

    public String getRegistrationValidationCharacter() {
        return registrationValidationCharacter;
    }

    public void setRegistrationValidationCharacter(String registrationValidationCharacter) {
        this.registrationValidationCharacter = registrationValidationCharacter;
    }

    public String getVin() {
        return vin;
    }

    public void setVin(String vin) {
        this.vin = vin;
    }

    public String getMakeCode() {
        return makeCode;
    }

    public void setMakeCode(String makeCode) {
        this.makeCode = makeCode;
    }

    public String getModelCode() {
        return modelCode;
    }

    public void setModelCode(String modelCode) {
        this.modelCode = modelCode;
    }

    public String getMakeInFull() {
        return makeInFull;
    }

    public void setMakeInFull(String makeInFull) {
        this.makeInFull = makeInFull;
    }

    public String getColour1Code() {
        return colour1Code;
    }

    public void setColour1Code(String colour1Code) {
        this.colour1Code = colour1Code;
    }

    public String getColour2Code() {
        return colour2Code;
    }

    public void setColour2Code(String colour2Code) {
        this.colour2Code = colour2Code;
    }

    public String getPropulsionCode() {
        return propulsionCode;
    }

    public void setPropulsionCode(String propulsionCode) {
        this.propulsionCode = propulsionCode;
    }

    public Integer getDesignedGrossWeight() {
        return designedGrossWeight;
    }

    public void setDesignedGrossWeight(Integer designedGrossWeight) {
        this.designedGrossWeight = designedGrossWeight;
    }

    public Integer getUnladenWeight() {
        return unladenWeight;
    }

    public void setUnladenWeight(Integer unladenWeight) {
        this.unladenWeight = unladenWeight;
    }

    public String getEngineNumber() {
        return engineNumber;
    }

    public void setEngineNumber(String engineNumber) {
        this.engineNumber = engineNumber;
    }

    public Integer getEngineCapacity() {
        return engineCapacity;
    }

    public void setEngineCapacity(Integer engineCapacity) {
        this.engineCapacity = engineCapacity;
    }

    public Integer getSeatingCapacity() {
        return seatingCapacity;
    }

    public void setSeatingCapacity(Integer seatingCapacity) {
        this.seatingCapacity = seatingCapacity;
    }

    public Date getManufactureDate() {
        return manufactureDate;
    }

    public void setManufactureDate(Date manufactureDate) {
        this.manufactureDate = manufactureDate;
    }

    public Date getFirstRegistrationDate() {
        return firstRegistrationDate;
    }

    public void setFirstRegistrationDate(Date firstRegistrationDate) {
        this.firstRegistrationDate = firstRegistrationDate;
    }

    public Boolean isSeriouslyDamaged() {
        return isSeriouslyDamaged;
    }

    public void setSeriouslyDamaged(Boolean isSeriouslyDamaged) {
        this.isSeriouslyDamaged = isSeriouslyDamaged;
    }

    public String getRecentV5DocumentNumber() {
        return recentV5DocumentNumber;
    }

    public void setRecentV5DocumentNumber(String recentV5DocumentNumber) {
        this.recentV5DocumentNumber = recentV5DocumentNumber;
    }

    public Boolean isVehicleNewAtFirstRegistration() {
        return isVehicleNewAtFirstRegistration;
    }

    public void setVehicleNewAtFirstRegistration(Boolean isVehicleNewAtFirstRegistration) {
        this.isVehicleNewAtFirstRegistration = isVehicleNewAtFirstRegistration;
    }

    public String getBodyTypeCode() {
        return bodyTypeCode;
    }

    public void setBodyTypeCode(String bodyTypeCode) {
        this.bodyTypeCode = bodyTypeCode;
    }

    public String getWheelPlanCode() {
        return wheelPlanCode;
    }

    public void setWheelPlanCode(String wheelPlanCode) {
        this.wheelPlanCode = wheelPlanCode;
    }

    public String getSvaEmissionStandard() {
        return svaEmissionStandard;
    }

    public void setSvaEmissionStandard(String svaEmissionStandard) {
        this.svaEmissionStandard = svaEmissionStandard;
    }

    public String getCtRelatedMark() {
        return ctRelatedMark;
    }

    public void setCtRelatedMark(String ctRelatedMark) {
        this.ctRelatedMark = ctRelatedMark;
    }

    public Integer getVehicleId() {
        return vehicleId;
    }

    public void setVehicleId(Integer vehicleId) {
        this.vehicleId = vehicleId;
    }

    public Integer getDvlaVehicleId() {
        return dvlaVehicleId;
    }

    public void setDvlaVehicleId(Integer dvlaVehicleId) {
        this.dvlaVehicleId = dvlaVehicleId;
    }

    public String getEuClassification() {
        return euClassification;
    }

    public void setEuClassification(String euClassification) {
        this.euClassification = euClassification;
    }

    public Integer getMassInServiceWeight() {
        return massInServiceWeight;
    }

    public void setMassInServiceWeight(Integer massInServiceWeight) {
        this.massInServiceWeight = massInServiceWeight;
    }

}

File: mot-api/module/DvsaEntities/src/DvsaEntities/Entity/DvlaVehicle.php
<?php

namespace DvsaEntities\Entity;

use Doctrine\ORM\Mapping as ORM;
use DvsaEntities\EntityTrait\CommonIdentityTrait;

/**
 * Vehicle information sent from DVLA.
 *
 * @ORM\Table(name="dvla_vehicle", options={"collate"="utf8_general_ci", "charset"="utf8", "engine"="InnoDB"})
 * @ORM\Entity(repositoryClass="DvsaEntities\Repository\DvlaVehicleRepository")
 */
class DvlaVehicle implements VehicleInterface
{
    use CommonIdentityTrait;

    /**
     * @var string
     *
     * @ORM\Column(name="registration", type="string", length=7, nullable=true)
     */
    private $registration;

    /**
     * @var string
     *
     * @ORM\Column(name="vin", type="string", length=20, nullable=true)
     */
    private $vin;

    /**
     * @var string
     *
     * @ORM\Column(name="make_code", type="string", length=5, nullable=false)
     */
    private $makeCode;

    /**
     * @var string
     *
     * @ORM\Column(name="make_in_full", type="string", length=20, nullable=true)
     */
    private $makeInFull;

    /**
     * @var string
     *
     * @ORM\Column(name="model_code", type="string", length=5, nullable=false)
     */
    private $modelCode;

    /**
     * @var string
     *
     * @ORM\Column(name="colour_1_code", type="string", length=1, nullable=true)
     */
    private $primaryColour;

    /**
     * @var string
     *
     * @ORM\Column(name="colour_2_code", type="string", length=1, nullable=true)
     */
    private $secondaryColour;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="manufacture_date", type="datetime", nullable=true)
     */
    private $manufactureDate;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="first_registration_date", type="date", nullable=true)
     */
    private $firstRegistrationDate;

    /**
     * @var string
     *
     * @ORM\Column(name="propulsion_code", type="string", length=2, nullable=true)
     */
    private $fuelType;

    /**
     * @var string
     *
     * @ORM\Column(name="body_type_code", type="string", length=5, nullable=false)
     */
    private $bodyType;

    /**
     * @var Vehicle
     *
     * @ORM\ManyToOne(targetEntity="DvsaEntities\Entity\Vehicle", fetch="LAZY")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="vehicle_id", referencedColumnName="id")
     * })
     */
    private $vehicle;

    /**
     * @var integer
     *
     * @ORM\Column(name="engine_capacity", type="integer", nullable=true)
     */
    private $cylinderCapacity;

    /**
     * Designed gross weight in kilos.
     *
     * @var integer
     *
     * @ORM\Column(name="designed_gross_weight", type="integer", nullable=true)
     */
    private $designedGrossWeight;

    /**
     * Unladen weight in kilos.
     *
     * @var integer
     *
     * @ORM\Column(name="unladen_weight", type="integer", nullable=true)
     */
    private $unladenWeight;

    /**
     * @var string
     *
     * @ORM\Column(name="engine_number", type="string", nullable=true)
     */
    private $engineNumber;

    /**
     * @var integer
     *
     * @ORM\Column(name="seating_capacity", type="integer", nullable=true)
     */
    private $seatingCapacity;

    /**
     * @var integer
     *
     * @ORM\Column(name="is_vehicle_new_at_first_registration", type="smallint", nullable=true)
     */
    private $newAtFirstReg;

    /**
     * @param Make|null
     */
    private $make;

    /**
     * @param Model|null
     */
    private $model;

    /**
     * @var ModelDetail|null
     */
    private $modelDetail = null;

    /**
     * @var integer
     *
     * @ORM\Column(name="recent_v5_document_number", type="integer", nullable=true)
     */
    private $v5DocumentNumber;

    /**
     * @param string $value body type code
     *
     * @return $this
     */
    public function setBodyType($value)
    {
        $this->bodyType = $value;

        return $this;
    }

    /**
     * @return string body type code
     */
    public function getBodyType()
    {
        return $this->bodyType;
    }

    /**
     * @param \DateTime $value
     *
     * @return DvlaVehicle
     */
    public function setFirstRegistrationDate($value)
    {
        $this->firstRegistrationDate = $value;

        return $this;
    }

    /**
     * @return \DateTime
     */
    public function getFirstRegistrationDate()
    {
        return $this->firstRegistrationDate;
    }


    /**
     * @param Integer $value
     * @return DvlaVehicle
     */
    public function setV5DocumentNumber($value)
    {
        $this->v5DocumentNumber = $value;
        return $this;
    }

    /**
     * @return Integer
     */
    public function getV5DocumentNumber()
    {
        return $this->v5DocumentNumber;
    }

    /**
     * First used date is either derived from DVLA vehicle data (Date of Manufacture and Date of First Registration) at
     * time of test registration or is entered by the NT.
     *
     * @return \DateTime
     */
    public function getFirstUsedDate()
    {
        return $this->isVehicleNewAtFirstRegistration() ?
            $this->getFirstRegistrationDate() : $this->getManufactureDate();
    }

    /**
     * @param string $value fuel type code
     *
     * @return $this
     */
    public function setFuelType($value)
    {
        $this->fuelType = $value;

        return $this;
    }

    /**
     * @return string
     */
    public function getFuelType()
    {
        return $this->fuelType;
    }

    /**
     * @param string $makeCode
     *
     * @return DvlaVehicle
     */
    public function setMakeCode($makeCode)
    {
        $this->makeCode = $makeCode;

        return $this;
    }

    /**
     * @return string
     */
    public function getMakeCode()
    {
        return $this->makeCode;
    }

    /**
     * @param Make|null $value
     *
     * @return DvlaVehicle
     */
    public function setMake(Make $value = null)
    {
        $this->make = $value;

        return $this;
    }

    /**
     * @return Make|null
     */
    public function getMake()
    {
        return $this->make;
    }

    /**
     * @return string|null
     */
    public function getMakeName()
    {
        return $this->make ? $this->make->getName() : null;
    }

    /**
     * @param string $modelCode
     *
     * @return DvlaVehicle
     */
    public function setModelCode($modelCode)
    {
        $this->modelCode = $modelCode;

        return $this;
    }

    /**
     * @return string
     */
    public function getModelCode()
    {
        return $this->modelCode;
    }

    /**
     * @param string $makeInFull
     *
     * @return DvlaVehicle
     */
    public function setMakeInFull($makeInFull)
    {
        $this->makeInFull = $makeInFull;

        return $this;
    }

    /**
     * @return string
     */
    public function getMakeInFull()
    {
        return $this->makeInFull;
    }
    /**
     * @return string|null
     */
    public function getModelName()
    {
        return $this->model ? $this->model->getName() : null;
    }

    /**
     * @param Model|null $model
     *
     * @return $this
     */
    public function setModel(Model $model = null)
    {
        $this->model = $model;

        return $this;
    }

    /**
     * @return Model|null
     */
    public function getModel()
    {
        return $this->model;
    }

    /**
     * @param ModelDetail|null $modelDetail
     *
     * @return $this
     */
    public function setModelDetail(ModelDetail $modelDetail = null)
    {
        $this->modelDetail = $modelDetail;

        return $this;
    }

    /**
     * @return ModelDetail|null
     */
    public function getModelDetail()
    {
        return $this->modelDetail;
    }

    /**
     * @param string $value colour code
     *
     * @return DvlaVehicle
     */
    public function setPrimaryColour($value)
    {
        $this->primaryColour = $value;

        return $this;
    }

    /**
     * @return string colour code
     */
    public function getPrimaryColour()
    {
        return $this->primaryColour;
    }

    /**
     * @param string $value colour code
     *
     * @return DvlaVehicle
     */
    public function setSecondaryColour($value)
    {
        $this->secondaryColour = $value;

        return $this;
    }

    /**
     * @return string colour code
     */
    public function getSecondaryColour()
    {
        return $this->secondaryColour;
    }

    /**
     * @param \DateTime $date
     *
     * @return Vehicle
     */
    public function setManufactureDate($date)
    {
        $this->manufactureDate = $date;

        return $this;
    }

    /**
     * @return \DateTime
     */
    public function getManufactureDate()
    {
        return $this->manufactureDate;
    }

    /**
     * @param string $value
     *
     * @return DvlaVehicle
     */
    public function setRegistration($value)
    {
        $this->registration = $value;

        return $this;
    }

    /**
     * @return string
     */
    public function getRegistration()
    {
        return $this->registration;
    }

    /**
     * @param $value
     *
     * @return DvlaVehicle
     */
    public function setVin($value)
    {
        $this->vin = $value;

        return $this;
    }

    /**
     * @return string
     */
    public function getVin()
    {
        return $this->vin;
    }

    /**
     * @param Vehicle $vehicle
     *
     * @return DvlaVehicle
     */
    public function setVehicle($vehicle)
    {
        $this->vehicle = $vehicle;

        return $this;
    }

    /**
     * @return Vehicle
     */
    public function getVehicle()
    {
        return $this->vehicle;
    }

    /**
     * @param int $cylinderCapacity
     *
     * @return DvlaVehicle
     */
    public function setCylinderCapacity($cylinderCapacity)
    {
        $this->cylinderCapacity = $cylinderCapacity;

        return $this;
    }

    /**
     * @return int
     */
    public function getCylinderCapacity()
    {
        return $this->cylinderCapacity;
    }

    /**
     * Set designedGrossWeight.
     *
     * @param integer $designedGrossWeight weight in kilos
     *
     * @return DvlaVehicle
     */
    public function setDesignedGrossWeight($designedGrossWeight)
    {
        $this->designedGrossWeight = $designedGrossWeight;

        return $this;
    }

    /**
     * Get designedGrossWeight.
     *
     * @return integer weight in kilos
     */
    public function getDesignedGrossWeight()
    {
        return $this->designedGrossWeight;
    }

    /**
     * Set unladenWeight.
     *
     * @param integer $unladenWeight weight in kilos
     *
     * @return DvlaVehicle
     */
    public function setUnladenWeight($unladenWeight)
    {
        $this->unladenWeight = $unladenWeight;

        return $this;
    }

    /**
     * Get unladen weight.
     *
     * @return integer weight in kilos
     */
    public function getUnladenWeight()
    {
        return $this->unladenWeight;
    }

    /**
     * @param string $engineNumber
     *
     * @return Vehicle
     */
    public function setEngineNumber($engineNumber)
    {
        $this->engineNumber = $engineNumber;

        return $this;
    }

    /**
     * @return string
     */
    public function getEngineNumber()
    {
        return $this->engineNumber;
    }

    /**
     * @param integer $seatingCapacity
     *
     * @return Vehicle
     */
    public function setSeatingCapacity($seatingCapacity)
    {
        $this->seatingCapacity = $seatingCapacity;

        return $this;
    }

    /**
     * @return integer
     */
    public function getSeatingCapacity()
    {
        return $this->seatingCapacity;
    }

    /**
     * @param integer $value
     *
     * @return Vehicle
     */
    public function setNewAtFirstReg($value)
    {
        $this->newAtFirstReg = $value;

        return $this;
    }

    /**
     * @return int
     */
    public function getNewAtFirstReg()
    {
        return $this->newAtFirstReg;
    }

    /**
     * @return bool
     */
    public function isVehicleNewAtFirstRegistration()
    {
        return (bool) $this->newAtFirstReg;
    }

    /**
     * Indicate based on instantiation the nature of this vehicles origins.
     *
     * @return bool
     */
    public function isDvla()
    {
        return true;
    }
}
